src/analysis/timeline/timelineCredView.test.js

```diff
@@ -0,0 +1,109 @@
+import {type FilteredTimelineCred} from "./filterTimelineCred";
+import {defaultWeights} from "../weights";
+
+describe("src/analysis/timeline/timelineCred", () => {
+  const credConfig: () => TimelineCredConfig = () => ({
+    scoreNodePrefix: NodeAddress.fromParts(["foo"]),
+    filterNodePrefixes: [NodeAddress.fromParts(["foo"])],
+    types: {nodeTypes: [], edgeTypes: []},
+  });
+
+  function exampleTimelineCred(): TimelineCred {
+    const startTimeMs = +new Date(2017, 0);
+    const endTimeMs = +new Date(2017, 6);
+    const boundaries = utcWeek.range(startTimeMs, endTimeMs);
+    const intervals = [];
+    for (let i = 0; i < boundaries.length - 1; i++) {
+      intervals.push({
+        startTimeMs: +boundaries[i],
+        endTimeMs: +boundaries[i + 1],
+      });
+    }
+    const users = [
+      ["starter", (x) => Math.max(0, 20 - x)],
+      ["steady", (_) => 4],
+      ["finisher", (x) => (x * x) / 20],
+      ["latecomer", (x) => Math.max(0, x - 20)],
+    ];
+
+    const graph = new Graph();
+    const addressToCred = new Map();
+    for (const [name, generator] of users) {
+      const address = NodeAddress.fromParts(["foo", name]);
+      graph.addNode({
+        address,
+        description: `[@${name}](https://github.com/${name})`,
+        timestampMs: null,
+      });
+      const scores = intervals.map((_unuesd, i) => generator(i));
+      addressToCred.set(address, scores);
+    }
+    const filteredTimelineCred: FilteredTimelineCred = {
+      intervals,
+      addressToCred,
+    };
+    const params = {alpha: 0.05, intervalDecay: 0.5, weights: defaultWeights()};
+    return new TimelineCred(graph, filteredTimelineCred, params, credConfig());
+  }
+
+  it("JSON serialization works", () => {
+    const json = exampleTimelineCred().toJSON();
+    const tc_ = TimelineCred.fromJSON(json, credConfig());
+    const tcv = new TimelineCredView(exampleTimelineCred());
+    const tcv_ = new TimelineCredView(tc_);
+    expect(tcv.credSortedNodes(NodeAddress.empty)).toEqual(
+      tcv_.credSortedNodes(NodeAddress.empty)
+    );
+  });
+
+  it("default interval works", () => {
+    const tc = exampleTimelineCred();
+    const tcv = new TimelineCredView(tc);
+    expect(tc.intervals()).toEqual(tcv.intervals());
+  });
+
+  it("single input interval works", () => {
+    const tc = exampleTimelineCred();
+    const startTimeMs = +new Date(2017, 1);
+    const inputInterval = {startTimeMs: startTimeMs, endTimeMs: startTimeMs};
+    const tcv = new TimelineCredView(tc, inputInterval);
+    expect(tcv.intervals().length).toEqual(1);
+  });
+
+  it("straddle input interval works", () => {
+    const tc = exampleTimelineCred();
+    const startTimeMs = +new Date(2017, 0, 14);
+    const endTimeMs = +new Date(2017, 0, 14 + 7 + 2);
+    const inputInterval = {startTimeMs: startTimeMs, endTimeMs: endTimeMs};
+    const tcv = new TimelineCredView(tc, inputInterval);
+    expect(tcv.intervals().length).toEqual(3);
```

Comment:
similar to the test above, it would be nice if the test was more specific about how the intervals should relate to the startTimeMs (e.g: startTimeMs is included within the first interval, endTimeMs is included within the last interval). However, I'm OK merging without this change.